admin.go

```diff
@@ -973,6 +982,13 @@ func handleConfig(w http.ResponseWriter, r *http.Request) error {
 		// we could consider setting up a sync.Pool for the summed
 		// hashes to reduce GC pressure.
 		w.Header().Set("Etag", makeEtag(r.URL.Path, hash))
+		_, err = w.Write(buf.Bytes())
+		if err != nil {
+			return APIError{HTTPStatus: http.StatusBadRequest, Err: err}
+		}
+
+		buf.Reset()
```

Comment:
We should move the `Reset` to when the buffer is retrieved (when client requests invalid config path it will not be recycled and rare chance response fails to write), and `defer` to return the buffer there.

Reply:
I agree, made the changes.